woocommerce/woocommerce-gutenberg-products-block
# Add eslint to find and fix code style issues

In this PR, I've added `eslint` and [the gutenberg `eslint` config,](https://github.com/WordPress/gutenberg/blob/master/eslint/config.js) then fixed all the errors this generated. The two changes I've made to the config is to drop an error to a warn on ["camelcase" issues](https://eslint.org/docs/rules/camelcase) (because we use camelcase properties a lot), and add a warning on [react/prop-types](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md), so that we define all props used in PropTypes.

Note: reviewing this might be easier to go commit-by-commit, rather than as one big diff.

**To test**

1. Run `npm install` to add the eslint dependencies
2. Run `npm run lint` to check for errors.
3. Expect: There are no errors (but there are 114 warnings– all `camelcase` and `react/prop-types`, these will be fixed as work on the plugin continues).

`lint` runs `lint:js`, which you can run directly with `npm run lint:js -- --quiet` to only show errors. We'll eventually add `lint:css` and `lint:php` [like wc-admin.](https://github.com/woocommerce/wc-admin/blob/master/package.json#L30-L35)

This PR also fixes #110 (since the key issue creates an error)
